lib/iris/tests/integration/test_netcdf__loadsaveattrs.py

```diff
@@ -364,30 +405,256 @@ def fetch_results(
+                # Return results as if all cubes had global=None
+                results = [
+                    [None]
+                    + [
+                        cube_attr_dict.get(attr_name, None)
+                        for cube_attr_dict in cube_attr_dicts
+                    ]
+                ]
+            else:
+                # Return a result-set for each occurring global value (possibly
+                # including a 'None').
+                global_values = set(
+                    cube.attributes.globals.get(attr_name, None)
                     for cube in cubes
-                    if cube.attributes.globals.get(attr_name, None)
-                    == globalval
+                )
+                results = [
+                    [globalval]
+                    + [
+                        cube.attributes.locals.get(attr_name, None)
+                        for cube in cubes
+                        if cube.attributes.globals.get(attr_name, None)
+                        == globalval
+                    ]
+                    for globalval in sorted(global_values, key=str)
                 ]
-                for globalval in sorted(global_values)
-            ]
         return results
 
 
+# Define all the testcases for different parameter input structures :
+#   - combinations of matching+differing, global+local params
+#   - these are interpreted differently for the 3 main test types : Load/Save/Roundtrip
+_MATRIX_TESTCASE_INPUTS = {
+    "case_singlevar_localonly": "G-La",
+    "case_singlevar_globalonly": "GaL-",
+    "case_singlevar_glsame": "GaLa",
+    "case_singlevar_gldiffer": "GaLb",
+    "case_multivar_same_noglobal": "G-Laa",
+    "case_multivar_same_sameglobal": "GaLaa",
+    "case_multivar_same_diffglobal": "GaLbb",
+    "case_multivar_differ_noglobal": "G-Lab",
+    "case_multivar_differ_diffglobal": "GaLbc",
+    "case_multivar_differ_sameglobal": "GaLab",
+    "case_multivar_1none_noglobal": "G-La-",
+    "case_multivar_1none_diffglobal": "GaLb-",
+    "case_multivar_1none_sameglobal": "GaLa-",
+    # Note: the multi-set input cases are more complex.
+    # These are encoded as *pairs* of specs, for 2 different files, or cubes with
+    #  independent global values.
+    # We assume that there can be nothing "special" about a var's interaction with
+    #  another one from the same (as opposed to the "other") file.
+    "case_multisource_gsame_lnone": ["GaL-", "GaL-"],
+    "case_multisource_gsame_lallsame": ["GaLa", "GaLa"],
+    "case_multisource_gsame_l1same1none": ["GaLa", "GaL-"],
+    "case_multisource_gsame_l1same1other": ["GaLa", "GaLb"],
+    "case_multisource_gsame_lallother": ["GaLb", "GaLb"],
+    "case_multisource_gsame_lalldiffer": ["GaLb", "GaLc"],
+    "case_multisource_gnone_l1same1none": ["G-La", "G-L-"],
+    "case_multisource_gnone_l1same1same": ["G-La", "G-La"],
+    "case_multisource_gnone_l1same1other": ["G-La", "G-Lb"],
```

Comment:
Ok I see this is slightly mis-named !
I guess this would make be more in keeping with the other names ...
```
    "case_multisource_gnone_l1one1none": ["G-La", "G-L-"],
    "case_multisource_gnone_l1one1same": ["G-La", "G-La"],
    "case_multisource_gnone_l1one1other": ["G-La", "G-Lb"],
```
Does that work for you ?

Reply:
Yes, this works for me. Sorry for the delay

Reply:
Thanks, that's a quick fix then !
https://github.com/SciTools/iris/pull/5410/commits/eea99d1d060cc64d6354021572f6fb89da4b2860